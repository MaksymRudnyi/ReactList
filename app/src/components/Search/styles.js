import styled from 'styled-components';

const Input = styled.input`
  border: 1px solid #cecece;
  padding: 15px;
  margin-bottom: 15px;
  display: flex;
  flex-direction: row;
  overflow: hidden;
  border-radius: 5px;
`;
Input.displayName = 'Input';

export {
	Input,
};
