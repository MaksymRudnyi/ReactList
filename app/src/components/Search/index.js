import React from 'react';
import PropTypes from 'prop-types';

import { Input } from './styles';

const Search = ({ term, onChange }) => (
  <Input
    type="text"
    value={term}
    onChange={onChange}
    placeholder="Company name (min. 3 chars)"
    autoFocus
  />
);

export default Search;

Search.propTypes = {
  term: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};
