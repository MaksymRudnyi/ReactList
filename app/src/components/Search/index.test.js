import React from 'react';
import toJson from 'enzyme-to-json';

import { shallow, mount } from 'enzyme';

import Search from './';

describe('Search', () => {
	let sut;

	it('should match snapshot', () => {
		sut = shallow(<Search onChange={()=>{}}/>);

		expect(toJson(sut)).toMatchSnapshot();
	});

	it('should match snapshot with predefined term', () => {
		sut = shallow(<Search term="search term" onChange={()=>{}} />);

		expect(toJson(sut)).toMatchSnapshot();
	});

	it('should call onChange handler when data is changed', () => {
		const onChange = jest.fn();

		sut = mount(<Search term="search term" onChange={onChange} />);

		const input = sut.find('input[type="text"]');

		input.simulate('change', { target: { value: 'company' } });

		expect(onChange).toHaveBeenCalled();
	})
});
