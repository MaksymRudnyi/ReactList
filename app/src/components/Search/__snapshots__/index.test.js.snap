// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot 1`] = `
<Input
  autoFocus={true}
  onChange={[Function]}
  placeholder="Company name (min. 3 chars)"
  type="text"
/>
`;

exports[`Search should match snapshot with predefined term 1`] = `
<Input
  autoFocus={true}
  onChange={[Function]}
  placeholder="Company name (min. 3 chars)"
  type="text"
  value="search term"
/>
`;
